File: offset-manager-remote/src/main/java/localoffsetmanager/service/dto/LastOffsetRecordBatchResponse.java
package localoffsetmanager.service.dto;

import jakarta.validation.constraints.NotEmpty;
import offsetmanager.domain.OffsetRecord;

import java.util.List;

public record LastOffsetRecordBatchResponse(
  @NotEmpty
  List<LastOffsetRecordResponse> lastOffsetRecords
) {

  public static LastOffsetRecordBatchResponse from(List<OffsetRecord> offsetRecords) {
    return new LastOffsetRecordBatchResponse(
      offsetRecords.stream()
        .map(LastOffsetRecordResponse::from)
        .toList()
    );
  }
}

File: offset-manager-api/src/main/java/offsetmanager/api/v1/dto/LastOffsetRecordBatchResponse.kt
package offsetmanager.api.v1.dto

import offsetmanager.domain.offset.OffsetRecord

data class LastOffsetRecordBatchResponse(
  val lastOffsetRecords: List<LastOffsetRecordResponse>
) {
  companion object {
    fun from(offsetRecords: List<OffsetRecord>): LastOffsetRecordBatchResponse {
      return LastOffsetRecordBatchResponse(
        offsetRecords.stream()
          .map<LastOffsetRecordResponse> { offsetRecord: OffsetRecord ->
            LastOffsetRecordResponse.from(
              offsetRecord
            )
          }
          .toList()
      )
    }
  }
}
